.github/workflows/ecr-shared-deploy-stage.yml

```diff
@@ -0,0 +1,95 @@
+# Stage ECS Container Deployment for Fargate & Lambda
+name: Stage Container Deploy
+
+on:
+  workflow_call:
+    inputs:
+      AWS_REGION:
+        required: true
+        type: string
+      GHA_ROLE:
+        required: true
+        type: string
+      ECR:
+        required: true
+        type: string
+      FUNCTION:
+        required: false
+        type: string
+      PREBUILD:
+        required: false
+        type: string
+
+# Set defaults
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  deploy:
+    name: Build and Deploy to ECR
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Check for Python or Ruby
+        # Check for the existence of files that set either the Python version or the Ruby version
+        # and capture this informaiton for later use in the job. These values are used for
+        # conditional steps below to setup/configure the build environment for the container.
+        run: |
+          if [[ -f ".python-version" ]]; then
+            echo "python_repo=1" >> $GITHUB_ENV
+          else [[ -f ".ruby-version" ]]; then
+            echo "ruby_repo=1" >> $GITHUB_ENV
+          fi
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCT_STAGE }}:role/${{ inputs.GHA_ROLE }}
```

Comment:
Is there any difference between the dev/stage workflows other than which secret they pull?

Reply:
Slightly. The dev workflow needs to do some extra work around tagging in order to make the takes that appear in AWS actually match the commit hashes that are visible in GitHub (see lines 60-76). This isn't necessary in the stage workflow because it's triggered by a push to `main` (where the dev workflow is triggered by a PR).

But, with a little extra work, it's likely pretty straightforward to pull these two shared workflows into one.